### PR TITLE
Re-implement detail reports in new coach dashboard

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
Re implements our pre-existing detail reports (exercise and quizzes) to fit into the data and routing patterns of the new coach dashboards.

Also implemented:
* Auto-scrolling of the attempt log list, to position the most recently selected attempt as close to the middle of the scroll box as possible.
* Title driving through a CoreBase prop (but this could well have been more easily set in the common exercise detail and quiz detail components, I did this in the case of quizzes, so that parallel implementations can be compared. Once we have the strings to drive more unique titles, the former implementation is probably to be preferred).

### Reviewer guidance
Generate exercise usage data, and quiz usage data.
Navigate to the root reports that should list learners, then manually add the learner id to the URL.
Investigate, check that it works!

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Gherkin stories have been updated
- [ ] Unit tests have been updated

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.rst

### CODE REVIEW

File: kolibri/core/assets/src/views/AttemptLogList.vue [hunk 1]
```diff
@@ -104,13 +103,27 @@
         '$coreTextDisabled',
       ]),
     },
+    mounted() {
+      this.$nextTick(() => {
+        this.scrollToSelectedAttemptLog(this.selectedQuestionNumber);
+      });
+    },
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/AttemptLogList.vue:110]: cool

File: kolibri/core/assets/src/views/CoreBase/index.vue [hunk 2]
```diff
@@ -193,6 +193,13 @@
         required: false,
         default: '',
       },
+      // Alternative to using metaInfo in a top level component to set the
+      // title of the HTML Document
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/CoreBase/index.vue:197]: Is this phrased as intended? 

I think _this_ strategy is "using `metaInfo` in a top level component...", whereas the other strategy was more like "using `metaInfo` in a subcomponent..."?

File: kolibri/plugins/coach/assets/src/constants/index.js [hunk 3]
```diff
@@ -14,6 +14,17 @@ export const PageNames = {
...
+  REPORTS_GROUP_REPORT_LESSON_EXERCISE_LEARNER_PAGE_ROOT:
+    'REPORTS_GROUP_REPORT_LESSON_EXERCISE_LEARNER_PAGE_ROOT',
+  REPORTS_LEARNER_ACTIVITY_EXERCISE_PAGE_ROOT: 'REPORTS_LEARNER_ACTIVITY_EXERCISE_PAGE_ROOT',
+  REPORTS_LEARNER_REPORT_LESSON_EXERCISE_PAGE_ROOT:
+    'REPORTS_LEARNER_REPORT_LESSON_EXERCISE_PAGE_ROOT',
+  REPORTS_LESSON_LEARNER_EXERCISE_PAGE_ROOT: 'REPORTS_LESSON_LEARNER_EXERCISE_PAGE_ROOT',
+  REPORTS_GROUP_LEARNER_REPORT_QUIZ_PAGE_ROOT: 'REPORTS_GROUP_LEARNER_REPORT_QUIZ_PAGE_ROOT',
+  REPORTS_GROUP_REPORT_QUIZ_LEARNER_PAGE_ROOT: 'REPORTS_GROUP_REPORT_QUIZ_LEARNER_PAGE_ROOT',
+  REPORTS_LEARNER_REPORT_QUIZ_PAGE_ROOT: 'REPORTS_LEARNER_REPORT_QUIZ_PAGE_ROOT',
+  REPORTS_QUIZ_LEARNER_PAGE_ROOT: 'REPORTS_QUIZ_LEARNER_PAGE_ROOT',
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/constants/index.js:27]: whoa

File: kolibri/plugins/coach/assets/src/views/common/LearnerExerciseReport.vue [hunk 4]
```diff
@@ -30,23 +102,120 @@
...
+        this.$emit('navigate', {
+          exerciseId: this.exercise.id,
+          learnerId: this.learnerId,
+          interactionIndex,
+          attemptId: this.attemptId,
+        });
+      },
+      toggleShowCorrectAnswer() {
+        this.showCorrectAnswer = !this.showCorrectAnswer;
+        this.$forceUpdate();
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/common/LearnerExerciseReport.vue:163]: interesting - why is this necessary? might be worth adding a comment?

File: kolibri/plugins/coach/assets/src/views/common/LearnerExerciseReport.vue [hunk 5]
```diff
@@ -1,27 +1,99 @@
...
+              :verb="VERBS.completed"
+              :icon="ICONS.star"
+            />
+            <LearnerProgressLabel
+              v-else-if="status === STATUSES.helpNeeded"
+              :count="1"
+              :verbosity="1"
+              :verb="VERBS.needHelp"
+              :icon="ICONS.star"
+            />
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/common/LearnerExerciseReport.vue]: these can all be replaced by

```html
<StatusSimple :status="status" />
```

File: kolibri/plugins/coach/assets/src/views/reports/ReportsLessonReportPage.vue [hunk 6]
```diff
@@ -30,15 +30,15 @@
                 :text="tableRow.title"
                 :to="classRoute(
                   'ReportsLessonExerciseLearnerListPage',
-                  { exerciseId: tableRow.content_id }
+                  { exerciseId: tableRow.node_id }
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/reports/ReportsLessonReportPage.vue]: this doesn't seem right.

I've been treating all URL params as `content_id`s, not `node_id`s

**Review by indirectlylit — COMMENTED** [review]
looks great!

the only major concern I have is passing node IDs into params, which I think might break assumptions in other parts of the handling logic

File: kolibri/plugins/coach/assets/src/views/common/LearnerExerciseReport.vue (hunk 4 above)
Comment (rtibbles) [reply kolibri/plugins/coach/assets/src/views/common/LearnerExerciseReport.vue:163]: This was copied from the old exercise detail report, so I do not actually know - there was no comment there either.

File: kolibri/plugins/coach/assets/src/views/reports/ReportsLessonReportPage.vue (hunk 6 above)
Comment (rtibbles) [reply kolibri/plugins/coach/assets/src/views/reports/ReportsLessonReportPage.vue]: Hrm, I made this edit early on to get stuff working. I think the major issue was that the node id is needed to pass into the exercise detail view handlers to get the specific exercise content, including the assessmentmetadata. Probable that I can find the node_id in another way.

File: kolibri/core/assets/src/views/CoreBase/index.vue (hunk 2 above)
Comment (rtibbles) [reply kolibri/core/assets/src/views/CoreBase/index.vue:197]: Well, the ultimate intent is that the metaInfo is not set on another component, but set as a prop on CoreBase - as I suggested in the PR description though, once implemented, I am not sure that implementing this prop buys us that much.

File: kolibri/plugins/coach/assets/src/views/common/LearnerExerciseReport.vue (hunk 4 above)
Comment (indirectlylit) [reply kolibri/plugins/coach/assets/src/views/common/LearnerExerciseReport.vue:163]: cool

File: kolibri/core/assets/src/views/CoreBase/index.vue (hunk 2 above)
Comment (indirectlylit) [reply kolibri/core/assets/src/views/CoreBase/index.vue:197]: Ah I think I understand what you're saying: because it's just as easy to drop `metaInfo` into `SomePage` as it is to pass a `pageTitle` prop into `CoreBase` from `SomePage`?

I agree.

I lean slightly towards a prop though because it uses standard Vue mechanisms. `metaInfo` feels a little magical and I wouldn't mind keeping it isolated to a single component.

**Review by indirectlylit — APPROVED** [review]
